presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeWindows.java

```diff
@@ -209,111 +224,156 @@ public void testIdenticalWindowSpecificationsDefaultFrame()
                 "SUM(quantity) OVER (PARTITION BY suppkey ORDER BY orderkey) sum_quantity_C, " +
                 "AVG(quantity) OVER (PARTITION BY suppkey ORDER BY orderkey ROWS BETWEEN CURRENT ROW AND UNBOUNDED FOLLOWING) avg_quantity_D, " +
                 "SUM(discount) OVER (PARTITION BY suppkey ORDER BY orderkey) sum_discount_C " +
                 "FROM lineitem";
 
+        PlanMatchPattern tableScan = tableScan(
+                "lineitem",
+                ImmutableMap.of(QUANTITY_ALIAS, "quantity",
+                        DISCOUNT_ALIAS, "discount",
+                        SUPPKEY_ALIAS, "suppkey",
+                        ORDERKEY_ALIAS, "orderkey"));
+
         assertUnitPlan(sql,
                 anyTree(
-                        window(ImmutableList.of(
-                                functionCall("avg", windowD, false, "quantity"),
-                                functionCall("sum", windowC, false, "discount"),
-                                functionCall("sum", windowC, false, "quantity")),
-                                any())));
+                        window(specificationD,
+                                ImmutableList.of(
+                                functionCall("avg", frameD, ImmutableList.of(QUANTITY_ALIAS)),
+                                functionCall("sum", UNSPECIFIED_FRAME, ImmutableList.of(DISCOUNT_ALIAS)),
+                                functionCall("sum", UNSPECIFIED_FRAME, ImmutableList.of(QUANTITY_ALIAS))),
+                                tableScan)));
     }
 
     @Test
     public void testNotMergeAcrossJoinBranches()
     {
+        String rOrderkeyAlias = "R_ORDERKEY";
+        String rShipdateAlias = "R_SHIPDATE";
+        String rQuantityAlias = "R_QUANTITY";
+
         @Language("SQL") String sql = "WITH foo AS (" +
                 "SELECT " +
                 "suppkey, orderkey, partkey, " +
-                "SUM(discount) OVER (PARTITION BY orderkey ORDER BY shipdate ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) a " +
+                "SUM(discount) OVER (PARTITION BY orderkey ORDER BY shipdate, quantity DESC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) a " +
                 "FROM lineitem WHERE (partkey = 272 OR partkey = 273) AND suppkey > 50 " +
                 "), " +
                 "bar AS ( " +
                 "SELECT " +
                 "suppkey, orderkey, partkey, " +
-                "AVG(quantity) OVER (PARTITION BY orderkey ORDER BY shipdate ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) b " +
+                "AVG(quantity) OVER (PARTITION BY orderkey ORDER BY shipdate, quantity DESC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) b " +
                 "FROM lineitem WHERE (partkey = 272 OR partkey = 273) AND suppkey > 50 " +
                 ")" +
                 "SELECT * FROM foo, bar WHERE foo.a = bar.b";
 
+        ExpectedValueProvider<WindowNode.Specification> leftSpecification = specification(
```

Comment:
Please extract this to a method or methods so that someone reading this test can get the general gist of it easier.